lib/chef/provider/package/chocolatey.rb

```diff
@@ -152,11 +147,15 @@ def check_resource_semantics!
         #
         # @return [String] full path of choco.exe
         def choco_exe
-          @choco_exe ||= ::File.join(
-            choco_install_path,
-            "bin",
-            "choco.exe"
-            )
+          @choco_exe ||=
+              # if this check is in #define_resource_requirements, it won't get
+              # run before choco.exe gets called from #load_current_resource.
+              exe_path = ::File.join(choco_install_path.to_s, "bin", "choco.exe")
+          if !::File.exist?(exe_path)
+            raise Chef::Exceptions::MissingLibrary, CHOCO_MISSING_MSG
+          else
+            exe_path
+          end
```

Comment:
could we do this a bit more like...

```
  @choco_exe ||=
    begin
      exe_path = ...
      raise unless exist?
      exe_path
   end
```

its both cleaner to read, and also if we've set choco_exe successfully we don't need to re-test if we should raise again the next time we hit it, and the return value buried inside the if is just a bit hard to read (much like this run on sentence...)